src/commands/admin/admin-subcommands/blacklist.ts

```diff
@@ -0,0 +1,216 @@
+
+/*
+ * A command for the administrators of Sushi Bot to use for
+ * blacklisting users
+ */
+
+// 7 days in seconds
+const BAN_DELETE_SECONDS: number = 604800;
+const name: string = "blacklist";
+
+export const command: Subcommand = {
+    // Command headers
+    data: new SlashCommandSubcommandBuilder()
+        .setName(name)
+        .setDescription("Blacklist a user across all Sushi Bot's servers")
+        .addUserOption(o =>
+            o.setName("user")
+                .setDescription("The user to blacklist from Sushi Bot")
+                .setRequired(true)
+        )
+        .addStringOption(o =>
+            o.setName("reason")
+                .setDescription("Why is this user is being blacklisted?")
+                .setRequired(true)
+        ),
+
+    // Command execution
+    async execute(ctx: ChatInputCommandInteraction): Promise<void> {
+        // The user to blacklist
+        const user: User = ctx.options.getUser("user");
+        const uid: string = user.id;
+        const reason: string = ctx.options.getString("reason");
+
+        await ctx.deferReply();
+
+        const bl: BlacklistT = await Blacklist.findById(process.env.BLACKLIST_ID);
+
+        // Should ideally not be null but in the off-chance,
+        // throw an error
+        if (!bl) {
+            await createBlacklist();
+        }
+
+        const users: Map<string, string> = bl.users;
+
+        // Do nothing if the user has already been blacklisted
+        if (users.get(uid)) {
+            await ctx.followUp("The specified user is already blacklisted");
+            return;
+        }
+
+        const millisToSecs: number = 1000;
+
+        // Ask for confirmation before blacklisting the target user
+        const embed: EmbedBuilder = new EmbedBuilder()
+            .setTitle(`Blacklist ${user.username}?`)
+            .setDescription("The following action will be difficult to reverse, are you sure?")
+            .setColor(RED)
+            .setAuthor({ name: "Blacklist", iconURL: ctx.client.user.avatarURL() })
+            .setThumbnail(user.avatarURL())
+            .addFields(
+                { name: "User ID", value: user.id, inline: true },
+                { name: "Created", value: `<t:${Math.floor(user.createdTimestamp / millisToSecs)}>`, inline: true },
+                { name: "Reason", value: reason }
+            );
+        
+        const yes: ButtonBuilder = new ButtonBuilder()
+            .setCustomId("yes")
+            .setLabel("Yes")
+            .setStyle(ButtonStyle.Success);
+        
+        const no: ButtonBuilder = new ButtonBuilder()
+            .setCustomId("no")
+            .setLabel("No")
+            .setStyle(ButtonStyle.Danger);
+
+        const row: ActionRowBuilder<ButtonBuilder> = new ActionRowBuilder<ButtonBuilder>()
+            .addComponents(yes, no);
+
+        const prompt: Message = await ctx.followUp({
+            embeds: [embed],
+            components: [row]
+        });
```

Comment:
And then asks the administrator whether to blacklist the specified user or not

This is because the blacklisting process will most likely be difficult to reverse